File: server/php/db/MysqlModel.php
<?php

class MysqlModel
{
    private $host = "127.0.0.1";
    private $name = "root";
    private $pwd = "root";
    private $port = "3306";

    static private $conn = null;
    private $mysqli = null;

    private $field = "title,categories,keywords,content";
    /**
     * 防止直接创建对象
     */
    private function __construct($dbname)
    {
        $conn = new mysqli($this->host, $this->name, $this->pwd, $dbname);
        $conn->query("SET NAMES utf8");//设置编码
         // 检测连接
        if ($conn->connect_error) {
            die("连接失败: " . $conn->connect_error);
        } else {
            $this->mysqli = $conn;
            return $conn;
        }
    }
    /**
     * 初始化类
     */
    static public function init($dbname)
    {
        if (!isset(self::$conn)) {
            self::$conn = new self($dbname);
        }
        return self::$conn;
    }
    /**
     * 获取所有数据
     */
    public function getAllData($table)
    {
        //查询语句
        $sql = "select id," . $this->field . " from " . $table;

        $result = $this->mysqli->query($sql);
        $data = array();
        while ($ros = $result->fetch_assoc()) {
            $data[] = $ros;
        }
        return $data;
    }
    /**
     * 获取一条数据
     */
    public function getOneData($table, $where=array())
    {
        if($where){
           $where_str="id='{$where['id']}'";
        }else{
            $where_str='id=0';
        }
        $sql = "select id," . $this->field . " from " . $table . " where " . $where_str;
        $data = array();
        $result = $this->mysqli->query($sql);
        while ($ros = $result->fetch_assoc()) {
            $data[] = $ros;
        }
        return $data;
    }
    /**
     * 数据添加
     */
    public function add($table, $data)
    {
        $title = $data['title'];
        $categories = $data['categories'];
        $keywords = $data['keywords'];
        $content = $data['content'];

        $sql = "insert into " . $table . " (" . $this->field . ") values('{$title}','{$categories}','{$keywords}','{$content}')";
        if ($this->mysqli->query($sql) === true) {
            return 1;
        } else {
            echo "Error: " . $sql . "<br>" . $conn->error;
        }

    }
    /**
     * 删除数据
     */
    public function del($table, $where)
    {
        $sql = "delete from " . $table . " where " . $where;
        if ($where) {
            if ($this->mysqli->query($sql) === true) {
                return 1;
            } else {
                echo "Error: " . $sql . "<br>" . $this->mysqli->error;
            }
        }
    }
    /**
     * 修改数据
     */
    public function edit($table, $where=array(), $data=array())
    {
        $str = '';
        foreach ($data as $k => $v) {
            if ($v == "") {
                continue;
            }
            $str .= $k . "='{$v}',";
        }
        //去除多余的，号
        $str = substr($str, 0, -1);
        if($where){
            $where_str="id='{$where['id']}'";
         }else{
             $where_str='id=0';
         }
        $sql = "update " . $table . " set " . $str . " where " . $where_str;

        if ($this->mysqli->query($sql) === true) {
            return 1;
        } else {
            echo "Error: " . $sql . "<br>" . $conn->error;
        }
    }
    /**
     * 防止用户复制对象
     */
    private function __clone()
    {

    }
}

File: server/python/db/mysqlModel.py
import mysql.connector
import pandas as pd


class MysqlModel():
    # mysql数据路类
    def __init__(self):
        # 方案一
        self._db = mysql.connector.connect(
            host="localhost",
            user="root",
            password="root",
            database="article",
            raise_on_warnings=True)
        # 方案二
        # self._db=_conn()

    def _conn(self):
        _cur = mysql.connector.connect(
            host="localhost", user="root", password="root", database="article")
        return _cur

    def getAllData(self, table):
        # 连接数据库
        _model = self._db.cursor(dictionary=True)
        # 构建sql语句
        sql = "select * from %s" % (table)
        # 执行sql语句
        _model.execute(sql)
        # 获取所有数据
        res = _model.fetchall()
        # 获取查询结果的字段描述
        # col_result = _model.description
        # 获取表头数据excel，需要import pandas as pd
        # df = pd.DataFrame(list(res))
        # 返回数据
        return res

    def getOneData(self, table, where):
        # 连接数据库
        _model = self._db.cursor(dictionary=True)
        # 构建sql语句
        sql = f"select * from {table} where id='{where['id']}'"
        # 执行sql语句
        _model.execute(sql)
        # 获取所有数据
        res = _model.fetchall()
        # 返回数据
        return res

    def add(self, table, data):
        # 连接数据库
        _model = self._db.cursor(dictionary=True)
        # 构建sql语句
        sql = f"insert into {table}(title,categories,keywords,content) values('{data['title']}','{data['categories']}','{data['keywords']}','{data['content']}')"
        # 执行sql语句
        _model.execute(sql)
        # 数据表有更新需要使用此语句
        self._db.commit()
        # 执行结果
        res = _model.rowcount
        # 返回数据
        return res

    def edit(self, table, where, data):
        # 连接数据库
        _model = self._db.cursor(dictionary=True)
        # 构建sql语句
        sql = f"update {table} set title='{data['title']}',categories='{data['categories']}',keywords='{data['keywords']}',content='{data['content']}' where id='{where['id']}'"
        # 执行sql语句
        _model.execute(sql)
        # 数据表有更新需要使用此语句
        self._db.commit()
        # 执行结果
        res = _model.rowcount
        # 返回数据
        return res

    def delete(self, table, where):
        # 连接数据库
        _model = self._db.cursor(dictionary=True)
        # 构建sql语句
        sql = f"delete from {table} where id={where['id']}"
        # 执行sql语句
        _model.execute(sql)
        # 数据表有更新需要使用此语句
        self._db.commit()
        # 执行结果
        res = _model.rowcount
        # 返回数据
        return res